src/keys.rs

```diff
@@ -281,77 +284,63 @@ impl IssuanceAuthorizingKey {
-    fn derive_inner(&self) -> pallas::Scalar {
-        to_scalar(PrfExpand::ZsaIsk.expand(&self.0))
+    /// Sign the provided message using the `IssuanceAuthorizingKey`.
+    /// Only supports signing of messages of length 32 bytes, since we will only be using it to sign 32 byte SIGHASH values.
+    pub fn try_sign(&self, msg: &[u8; 32]) -> Result<Signature, schnorr::Error> {
+        schnorr::SigningKey::from(self.0).sign_prehash(msg)
     }
+}
 
-    /// Sign the provided message using the `IssuanceAuthorizingKey`.
-    pub fn sign(
-        &self,
-        rng: &mut (impl RngCore + CryptoRng),
-        msg: &[u8],
-    ) -> redpallas::Signature<IssuanceAuth> {
-        conditionally_negate(self.derive_inner()).sign(rng, msg)
+impl Debug for IssuanceAuthorizingKey {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("IssuanceAuthorizingKey")
+            .field(&self.0.to_bytes())
+            .finish()
     }
 }
 
 /// A key used to validate issuance authorization signatures.
 ///
 /// Defined in [ZIP 227: Issuance of Zcash Shielded Assets § Issuance Key Generation][IssuanceZSA].
-/// Note that this is $\mathsf{ik}^\mathbb{P}$, which by construction is equivalent to
-/// $\mathsf{ik}$ but stored here as a RedPallas verification key.
 ///
 /// [IssuanceZSA]: https://qed-it.github.io/zips/zip-0227#issuance-key-derivation
-#[derive(Debug, Clone, PartialOrd, Ord)]
-pub struct IssuanceValidatingKey(VerificationKey<IssuanceAuth>);
+#[derive(Debug, Clone)]
+pub struct IssuanceValidatingKey(schnorr::VerifyingKey);
 
 impl From<&IssuanceAuthorizingKey> for IssuanceValidatingKey {
     fn from(isk: &IssuanceAuthorizingKey) -> Self {
-        IssuanceValidatingKey((&(conditionally_negate(isk.derive_inner()))).into())
-    }
-}
-
-impl From<&IssuanceValidatingKey> for pallas::Point {
-    fn from(issuance_validating_key: &IssuanceValidatingKey) -> pallas::Point {
-        pallas::Point::from_bytes(&(&issuance_validating_key.0).into()).unwrap()
+        IssuanceValidatingKey(*schnorr::SigningKey::from(isk.0).verifying_key())
     }
 }
 
 impl PartialEq for IssuanceValidatingKey {
     fn eq(&self, other: &Self) -> bool {
-        <[u8; 32]>::from(&self.0).eq(&<[u8; 32]>::from(&other.0))
+        self.to_bytes().eq(&other.to_bytes())
     }
 }
 
 impl Eq for IssuanceValidatingKey {}
 
 impl IssuanceValidatingKey {
-    /// Converts this spend validating key to its serialized form,
-    /// I2LEOSP_256(ik).
+    /// Converts this issuance validating key to its serialized form,
+    /// in big-endian order as defined in BIP 340.
     pub fn to_bytes(&self) -> [u8; 32] {
-        // This is correct because the wrapped point must have ỹ = 0, and
-        // so the point repr is the same as I2LEOSP of its x-coordinate.
-        <[u8; 32]>::from(&self.0)
+        self.0.to_bytes().try_into().unwrap() // This cannot fail since the issuance validating key is exactly 32 bytes.
```

Comment:
same for comment, even better, move to above function header for both.

Reply:
Moved to above function header